### ISSUE TITLE
only works on new page load

### ISSUE THREAD

**Issue by jsjoeio** [wesbos/who-blue#2]
See video:

https://user-images.githubusercontent.com/3806031/201135148-ca06d6ef-121f-44e4-b9cb-c4fa9e36e9b9.mov

**Comment by wesbos** [comment]
yeah def PR this :) Didnt even think about how its a SPA. Some sort of history change event

**Comment by jsjoeio** [comment]
Leaving some notes here...

I see the main function called is `go`

https://github.com/wesbos/who-blue/blob/main/scripts/content.js#L31-L40

Basically we need to "subscribe" to some sort of history change event. Maybe this one: 

> Listening to [hashchange](https://developer.mozilla.org/docs/Web/API/HashChangeEvent), [popstate](https://developer.mozilla.org/docs/Web/Events/popstate) events:

```js
window.addEventListener('hashchange', e => {
  console.log('URL hash changed', e);
  doSomething();
});
window.addEventListener('popstate', e => {
  console.log('State changed', e);
  doSomething();
});
```

[Source](https://stackoverflow.com/questions/2844565/is-there-a-javascript-jquery-dom-change-listener/39508954#39508954)

**Comment by jsjoeio** [comment]
I tested both of those by putting them in the `console` on twitter.com and then navigating to a new profile via Twitter Search but neither logged. Ditching that approach.

Maybe one of these: 
- https://stackoverflow.com/questions/34957319/how-to-listen-for-url-change-with-chrome-extension
- https://medium.com/@softvar/making-chrome-extension-smart-by-supporting-spa-websites-1f76593637e8

**Comment by Sync271** [comment]
> Leaving some notes here...
> 
> I see the main function called is `go`
> 
> https://github.com/wesbos/who-blue/blob/main/scripts/content.js#L31-L40
> 
> Basically we need to "subscribe" to some sort of history change event. Maybe this one:
> 
> > Listening to [hashchange](https://developer.mozilla.org/docs/Web/API/HashChangeEvent), [popstate](https://developer.mozilla.org/docs/Web/Events/popstate) events:
> 
> ```js
> window.addEventListener('hashchange', e => {
>   console.log('URL hash changed', e);
>   doSomething();
> });
> window.addEventListener('popstate', e => {
>   console.log('State changed', e);
>   doSomething();
> });
> ```
> 
> [Source](https://stackoverflow.com/questions/2844565/is-there-a-javascript-jquery-dom-change-listener/39508954#39508954)

Hey, so I did test both `popstate` and `hashchange`, `popstate` works only when you are going back and forth between pages using "back" and "forward" buttons (realised this after making the PR 😓) but what we can actually do is a combination of `popstate` and `click` to run when you click on a profile/page.

**Comment by JeffreyATW** [comment]
Might also consider doing this on `scroll`, as new elements appear as you do so.

**Comment by davwheat** [comment]
Potentially a `MutationObserver` looking for SVGs where `aria-label` is `Verified account` might be a better option?
